crates/task-impls/src/quorum_vote.rs

```diff
@@ -250,19 +250,47 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumVoteTaskState<TYPES, I
         view_number: TYPES::Time,
         event_receiver: Receiver<Arc<HotShotEvent<TYPES>>>,
         event_sender: &Sender<Arc<HotShotEvent<TYPES>>>,
+        event: Option<Arc<HotShotEvent<TYPES>>>,
     ) {
         if self.vote_dependencies.get(&view_number).is_some() {
             return;
         }
-        let deps = vec![
-            self.create_event_dependency(
-                VoteDependency::QuorumProposal,
-                view_number,
-                event_receiver.clone(),
-            ),
-            self.create_event_dependency(VoteDependency::Dac, view_number, event_receiver.clone()),
-            self.create_event_dependency(VoteDependency::Vid, view_number, event_receiver),
-        ];
+
+        let mut quorum_proposal_dependency = self.create_event_dependency(
+            VoteDependency::QuorumProposal,
+            view_number,
+            event_receiver.clone(),
+        );
+        let mut dac_dependency =
+            self.create_event_dependency(VoteDependency::Dac, view_number, event_receiver.clone());
+        let mut vid_dependency =
+            self.create_event_dependency(VoteDependency::Vid, view_number, event_receiver);
+
+        // If we have an event provided to us
+        if let Some(event) = event {
+            // Match on the type of event
+            if let HotShotEvent::VoteNow(_, vote_dependency_data) = event.as_ref() {
+                // Then, mark all the dependencies as completed. Here, we short-circuit the entire dependency chain by marking
+                // *all* of the events as completed so that way this starts and ends immediately (thus voting).
+                quorum_proposal_dependency.mark_as_completed(
+                    HotShotEvent::QuorumProposalValidated(
+                        vote_dependency_data.quorum_proposal.clone(),
+                        vote_dependency_data.leaf.clone(),
+                    )
+                    .into(),
+                );
+                vid_dependency.mark_as_completed(
+                    HotShotEvent::VIDShareValidated(vote_dependency_data.disperse_proposal.clone())
+                        .into(),
+                );
+                dac_dependency.mark_as_completed(
+                    HotShotEvent::DACertificateValidated(vote_dependency_data.da_cert.clone())
+                        .into(),
+                );
+            };
+        }
```

Comment:
Similar question as in `quorum_proposal.rs` about whether we can change the dependency setting to avoid marking all dependencies.